java/team25core/DeadReckonTaskWithIMU.java

```diff
@@ -0,0 +1,548 @@
+        imuCalibTlm = telemetry.addData("Calib", imu.getCalibrationStatus().toString());
+        imuHeadingTlm = telemetry.addData("Heading", formatAngle(angles.angleUnit, angles.firstAngle));
+        imuYawRateTlm = telemetry.addData("YawRate", formatAngle(angles.angleUnit, angles.firstAngle));
+
+        imuRollTlm = telemetry.addData("Roll", formatAngle(angles.angleUnit, angles.secondAngle));
+        imuPitchTlm = telemetry.addData("Pitch", formatAngle(angles.angleUnit, angles.thirdAngle));
+        imuGravTlm = telemetry.addData("Grav", gravity.toString());
+        // whereAmIGyro = telemetry.addData("whereAmIGyro", "initTelemetry" );
+        telemetry.setMsTransmissionInterval(100);
+    }
+
+    @Override
+    public void start()
+    {
+        RobotLog.i(TAG, "Start");
+    }
+
+    @Override
+    public void stop()
+    {
+        RobotLog.i(TAG, "Stop");
+        drivetrain.stop();
+    }
+
+    public void useSmoothStart(boolean on)
+    {
+        smoothStart = on;
+    }
+
+    public void setTarget(DeadReckonPath.Segment segment)
+    {
+        switch (segment.type) {
+        case STRAIGHT:
+        case RIGHT_DIAGONAL:
+        case LEFT_DIAGONAL:
+        case SIDEWAYS:
+            drivetrain.setTargetInches(segment.distance);
+            break;
+        case TURN:
+            drivetrain.setTargetRotation(segment.distance);
+            break;
+        }
+    }
+
+    public boolean hitTarget()
+    {
+        if (drivetrain.isBusy()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public void disableSensors() {
+        sensorsInstalled = SensorsInstalled.SENSORS_NONE;
+    }
+
+    protected void setupWaitState(DeadReckonPath.Segment segment, boolean sendEvent)
+    {
+        if (sendEvent == true) {
+            robot.queueEvent(new DeadReckonEvent(this, EventKind.PAUSING, num));
+        }
+        segment.state = WAIT;
+        timer = new ElapsedTime(ElapsedTime.Resolution.MILLISECONDS);
+    }
+
+    public void getIMUValues() {
+        heading = -imu.getAngularOrientation().firstAngle;
+        roll = -imu.getAngularOrientation().secondAngle;
+        pitch = -imu.getAngularOrientation().thirdAngle;
+        yawRate = -imu.getAngularVelocity().yRotationRate;
+        drivetrain.setCurrentYaw(heading);
+        drivetrain.setCurrentYawRate(yawRate);
+    }
+
+    public void displayTelemetry() {
+        //telemetry.setAutoClear(false);
+        //imuStatusTlm.setValue(imu.getSystemStatus().toString());
+        // imuCalibTlm.setValue(imu.getCalibrationStatus().toString());
+
+     //   i = i++;
+     //   whereAmIGyro.setValue("displayTelemetry" + i);
+
+        this.imuHeadingTlm.setValue(heading);
+
+        this.imuRollTlm.setValue(roll);
+
+        this.imuPitchTlm.setValue(pitch);
+
+        this.imuYawRateTlm.setValue( yawRate );
+        //imuGravTlm.setValue(gravity.toString());
+
+
+    }
+
+    @Override
+    public boolean timeslice()
```

Comment:
Maybe, I'm missing something, and if so, please educate me, but this doesn't appear to actually be using the IMU to adjust the path the robot is moving.

All I initially see is the IMU being used for telemetry.  Which if that's what you want should be its own, completely independent task.  What we have here is a very large copy/paste of the original DeadReckonTask.  I'd like someone to chime in on the thought process behind this implementation.